luatest/assertions.lua

```diff
@@ -15,6 +15,16 @@ local prettystr_pairs = pp.tostring_pair
 
 local M = {}
 
+local mt = {}
+setmetatable(M, mt)
+
+mt.xfail = false
+function mt.is_xfail()
+    local xfail = mt.xfail
+    mt.xfail = false
+    return xfail
+end
+
```

Comment:
I don't like the idea of using metatables as private storage. Look, there's `M.private` definition two lines below. Let's put the `is_xfail` function there?